include/deal.II/grid/tria_accessor.templates.h

```diff
@@ -2040,6 +2040,141 @@ TriaAccessor<structdim, dim, spacedim>::diameter () const
 
 
 
+namespace
+{
+  /**
+   * For a given center/radius, see if any points are outside.
+   * If so, enlarge the ball and return true;
+   * Otherwise, return false;
+   */
+  template <int structdim, int dim, int spacedim>
+  inline
+  bool
+  enlarge_sphere(std::array<bool,GeometryInfo<structdim>::vertices_per_cell> &vertices_within_ball,
+                 const TriaAccessor<structdim, dim, spacedim> &cell,
+                 Point<spacedim> &center,
+                 double &radius)
+  {
+    for (unsigned int v = 0; v < GeometryInfo<structdim>::vertices_per_cell; ++v)
```

Comment:
@vishalkenchan after adding 2d/3D tests along the lines of what we discussed yesterday, please move this code directly where it is called. We should be fine to call it only once and have a single loop over all vertices. Then each time we find a vertex outside the ball -- enlarge. We should not need to break the loop and restart it.